src/eigen/lobpcg_hyper_impl.jl

```diff
@@ -43,20 +43,96 @@
 vprintln(args...) = nothing
 
 using LinearAlgebra
-using BlockArrays # used for the `mortar` command which makes block matrices
+using CUDA
+using GPUArrays
 
-# when X or Y are BlockArrays, this makes the return value be a proper array (not a BlockArray)
-function array_mul(X::AbstractArray{T}, Y) where T
-    Z = Array{T}(undef, size(X, 1), size(Y, 2))
-    mul!(Z, X, Y)
+
+# For now, BlockVector can store arrays of different types (for example, an element of type views and one of type Matrix). Maybe for performance issues it should only store arrays of the same type?
+
+struct BlockVector
+    blocks::Tuple
+    size::Tuple{Int64,Int64}
+end
```

Comment:
that's a bit too generic I think. For our purposes you can get by with a simplified version. Check out the history of this file, we used to have a "bmul" thing that did the job reasonably well and was a bit simpler